lib/editor/components/timetable/CalendarSelect.js

```diff
@@ -111,6 +119,18 @@ export default class CalendarSelect extends Component<Props> {
         }))
       : []
 
+    // Add exception based calendars to the list
+    // TODO: trip counts (total v pattern)
+    const exceptionBasedCalendarOptions = exceptionBasedCalendars && exceptionBasedCalendars.map(exception => ({
+      label: exception.name || exception.id,
+      value: exception.service_id,
+      type: 'exception-based',
+      service_id: exception.service_id // For an exception based schedule the custom schedule should only have one entry.
+    }))
+
+    // $FlowFixMe: we need two types of options
+    calendarOptions = calendarOptions.concat(exceptionBasedCalendarOptions)
```

Comment:
Can we keep this as a const but define it here?

Reply:
Yep, changed that in https://github.com/ibi-group/datatools-ui/pull/999/commits/96bff40f3ba89ac45676aa84fedfed73aab79494